File: app/src/main/java/es/etologic/mahjongscoring2/app/custom_views/VerticalTextView.java
package es.etologic.mahjongscoring2.app.custom_views;

import android.content.Context;
import android.graphics.Canvas;
import android.text.TextPaint;
import android.util.AttributeSet;

public class VerticalTextView extends android.support.v7.widget.AppCompatTextView {


    public VerticalTextView(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @SuppressWarnings("SuspiciousNameCombination")
    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        super.onMeasure(heightMeasureSpec, widthMeasureSpec);
        setMeasuredDimension(getMeasuredHeight(), getMeasuredWidth());
    }

    @Override
    protected void onDraw(Canvas canvas) {
        TextPaint textPaint = getPaint();
        textPaint.setColor(getCurrentTextColor());
        textPaint.drawableState = getDrawableState();

        canvas.save();

        canvas.translate(getWidth(), 0);
        canvas.rotate(90);

        canvas.translate(getCompoundPaddingLeft(), getExtendedPaddingTop());

        getLayout().draw(canvas);
        canvas.restore();
    }
}

File: app/src/main/java/com/etologic/mahjongscoring2/app/custom_views/VerticalTextView.kt
package com.etologic.mahjongscoring2.app.custom_views

import android.content.Context
import android.graphics.Canvas
import android.util.AttributeSet

class VerticalTextView(context: Context, attrs: AttributeSet) : androidx.appcompat.widget.AppCompatTextView(context, attrs) {
    
    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        super.onMeasure(heightMeasureSpec, widthMeasureSpec)
        setMeasuredDimension(measuredHeight, measuredWidth)
    }
    
    override fun onDraw(canvas: Canvas) {
        val textPaint = paint
        textPaint.color = currentTextColor
        textPaint.drawableState = drawableState
        
        canvas.save()
        
        canvas.translate(width.toFloat(), 0f)
        canvas.rotate(90f)
        
        canvas.translate(compoundPaddingLeft.toFloat(), extendedPaddingTop.toFloat())
        
        layout.draw(canvas)
        canvas.restore()
    }
}